src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/ECDiffieHellmanAndroid.Derive.cs

```diff
@@ -0,0 +1,197 @@
+                if (string.IsNullOrEmpty(hashAlgorithm.Name))
+                    throw new ArgumentException(SR.Cryptography_HashAlgorithmNameNullOrEmpty, nameof(hashAlgorithm));
+
+                ThrowIfDisposed();
+
+                return ECDiffieHellmanDerivation.DeriveKeyFromHmac(
+                    otherPartyPublicKey,
+                    hashAlgorithm,
+                    hmacKey,
+                    secretPrepend,
+                    secretAppend,
+                    (pubKey, hasher) => DeriveSecretAgreement(pubKey, hasher));
+            }
+
+            public override byte[] DeriveKeyTls(ECDiffieHellmanPublicKey otherPartyPublicKey, byte[] prfLabel, byte[] prfSeed)
+            {
+                if (otherPartyPublicKey == null)
+                    throw new ArgumentNullException(nameof(otherPartyPublicKey));
+                if (prfLabel == null)
+                    throw new ArgumentNullException(nameof(prfLabel));
+                if (prfSeed == null)
+                    throw new ArgumentNullException(nameof(prfSeed));
+
+                ThrowIfDisposed();
+
+                return ECDiffieHellmanDerivation.DeriveKeyTls(
+                    otherPartyPublicKey,
+                    prfLabel,
+                    prfSeed,
+                    (pubKey, hasher) => DeriveSecretAgreement(pubKey, hasher));
+            }
+
+            /// <summary>
+            /// Get the secret agreement generated between two parties
+            /// </summary>
+            private byte[]? DeriveSecretAgreement(ECDiffieHellmanPublicKey otherPartyPublicKey, IncrementalHash? hasher)
+            {
+                Debug.Assert(otherPartyPublicKey != null);
+
+                // Ensure that this ECDH object contains a private key by attempting a parameter export
+                // which will throw an OpenSslCryptoException if no private key is available
+                ECParameters thisKeyExplicit = ExportExplicitParameters(true);
+                bool thisIsNamed = Interop.AndroidCrypto.EcKeyHasCurveName(_key.Value);
+                ECDiffieHellmanAndroidPublicKey? otherKey = otherPartyPublicKey as ECDiffieHellmanAndroidPublicKey;
+                bool disposeOtherKey = false;
+
+                if (otherKey == null)
+                {
+                    disposeOtherKey = true;
+
+                    ECParameters otherParameters =
+                        thisIsNamed
+                            ? otherPartyPublicKey.ExportParameters()
+                            : otherPartyPublicKey.ExportExplicitParameters();
+
+                    otherKey = new ECDiffieHellmanAndroidPublicKey(otherParameters);
+                }
+
+                bool otherIsNamed = otherKey.HasCurveName;
+
+                SafeEcKeyHandle? ourKey = null;
+                SafeEcKeyHandle? theirKey = null;
+                byte[]? rented = null;
+                // Calculate secretLength in bytes.
+                int secretLength = (KeySize + 7) / 8;
```

Comment:
For all the places in managed code that do this (F3 shows me two instances of "+ 7" in this PR, but some might be folded away), I got feedback before that this formula (burned into my brain) requires too much thinking.  So there's a helper method for it: AsymmetricAlgorithmHelpers.BitsToBytes(bitLength) => byteLength.

Reply:
One occurrence I believe is in native code, so I'll leave that one as is. I'll change the other one to use the helper.